File: backend/scripts/check_images.py
import re
import urllib.request
import json

with open("seed.js", "r", encoding="utf-8") as f:
    content = f.read()

urls = re.findall(r'image:\s*"(https://images\.unsplash\.com/photo-[^?]+)[^"]*"', content)

for url in urls:
    try:
        photo_id = url.split("photo-")[1]
        html_url = f"https://unsplash.com/photos/{photo_id}"
        req = urllib.request.Request(html_url, headers={'User-Agent': 'Mozilla/5.0'})
        html = urllib.request.urlopen(req).read().decode('utf-8')
        title_match = re.search(r'<title>(.*?)</title>', html)
        print(f"ID: {photo_id} | Title: {title_match.group(1).split('|')[0].strip() if title_match else 'N/A'}")
    except Exception as e:
        print(f"ID: {url.split('photo-')[1]} | Error: {e}")

File: backend/scripts/check_images.js
const fs = require('fs');
const https = require('https');

const content = fs.readFileSync('seed.js', 'utf8');
const urls = [...content.matchAll(/image:\s*"(https:\/\/images\.unsplash\.com\/photo-[^?]+)[^"]*"/g)].map(m => m[1]);

async function checkUrl(url) {
    const photo_id = url.split("photo-")[1];
    const html_url = `https://unsplash.com/photos/${photo_id}`;
    
    return new Promise((resolve) => {
        https.get(html_url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => {
                const match = data.match(/<title>(.*?)<\/title>/);
                const title = match ? match[1].split('|')[0].trim() : 'N/A';
                console.log(`ID: ${photo_id} | Title: ${title}`);
                resolve();
            });
        }).on('error', (err) => {
            console.log(`ID: ${photo_id} | Error: ${err.message}`);
            resolve();
        });
    });
}

async function run() {
    for (const url of urls) {
        await checkUrl(url);
    }
}

run();
